src/book/03-guides/01-main-codebase/02-maintaining-subsystems.mdx

```diff
@@ -0,0 +1,121 @@
+
+### The Visual Mesh
+
+#### Training and Testing
+
+Go to the [NUbook Visual Mesh Getting Started guide](/guides/tools/visualmesh) to find out how to train and test a network, with an example dummy dataset.
+
+#### Exporting Configuration
+
+The resulting networking should be exported to a [yaml](https://yaml.org/) file and added to the NUbots codebase.
+
+Example yaml files can be found [in the Visual Mesh repository](https://github.com/Fastcode/VisualMesh/blob/master/example/model.yaml) and [in the NUbots repository](https://github.com/NUbots/NUbots/blob/master/module/vision/VisualMesh/data/config/VisualMesh.yaml). Assuming such a file is in the output directory, run
+
+```bash
+./mesh.py export [output_dir]
+```
+
+to export the weights to the file.
+
+Replace the NUbots repository `VisualMesh.yaml` configuration with this configuration file. Ensure you create a new branch so you can make a PR and get any changes merged in. View the [Git Guide](/guides/general/git) for information on using Git.
+
+### Camera Calibration
+
+The vision system cannot work optimally if the cameras are not calibrated correctly. [The input page](/system/subsystems/input) describes the camera parameters that can be calibrated.
+
+An automatic camera calibration tool is available in the NUbots repository. See the [camera calibration guide](guides/main/camera-calibration#page-content) to find out how to use this tool.
+
+### Testing
+
+After updating the Visual Mesh in the NUbots repository, it should be tested before merging.
+
+1. Build the code, ensuring `ROLE_visualmesh` is set to `ON` in `./b configure -i`. Go to the [Getting Started page](/guides/main/getting-started) to find out more on how to build the code and install to a robot. Ensure the new configuration file has been installed. Check out the [Build System page](/system/foundations/build-system#install) to find out more about options when installing onto the robot.
+
+2. When your new Visual Mesh is installed onto the robot, connect to the robot using
+
+   ```bash
+   ssh nubots@<address>
+   ```
+
+3. Ensure NUsight is on.
+
+   ```bash
+   nano config/NetworkForwarder.yaml
+   ```
+
+   Turn `vision object` and `compressed images` on. Run NUsight using `yarn prod` and navigate to the NUsight page in your browser. More on NUsight can be found on [the NUsight NUbook page](/system/tools/nusight).
+
+4. Run the visualmesh role
+
+   ```bash
+   ./visualmesh
+   ```
+
+5. Wait for the cameras to load and then watch the Vision tab in NUsight. To determine if the output is correct, consult the [vision page](/system/subsystems/vision) for images of the expected output.
+
+If you would like to see the Visual Mesh output in NUsight, you will need to log the data and run it back in NUsight using DataPlayback, since the data is too large to send over a network. See the [DataLogging and DataPlayback guide](/guides/main/data-recording-playback), considering the following:
```

Comment:
"considering the following"? I don't understand

Reply:
I'll try to find a better phrase. It's like 'follow the steps in that guide, but use these specific things for all the general things in the guide'.